documentation/Makefile

```diff
@@ -2,7 +2,9 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS   ?= -vn
+SPHINXOPTS   ?= -v -W -E
```

Comment:
I should add that the addition of these means that the doc build breaks on my machine. Seems ok on CI though. On current master, I get the same error about the dot file but it doesn't break the build as we don't have warnings as errors there. I don't see this as a blocker though I would be happy to understand it better at some point.

Reply:
If you're getting it on master branch too then there's something wrong, but I can't reproduce it. If you could provide the full traceback, I can provide better feedback (other than "it works for me").

Did you try doing `make clean` before rebuilding the docs? Maybe something went awry in the last attempt, and it is preventing incremental attempts. I still suspect a path problem (maybe related to using doxygen 1.8.17 output).

Reply:
I was finally able to reproduce this with a fresh install (of everything) on ubuntu (newly dual booted).

<details><summary>full traceback error</summary>

```
Traceback (most recent call last):
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/cmd/build.py", line 284, in build_main
    app.build(args.force_all, filenames)
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/application.py", line 337, in build
    self.builder.build_update()
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 294, in build_update
    self.build(to_build,
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 307, in build
    with logging.pending_warnings():
  File "/usr/local/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/util/logging.py", line 225, in pending_warnings
    memhandler.flushTo(logger)
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/util/logging.py", line 190, in flushTo
    logger.handle(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1634, in handle
    self.callHandlers(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 1696, in callHandlers
    hdlr.handle(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 964, in handle
    rv = self.filter(record)
  File "/usr/local/lib/python3.10/logging/__init__.py", line 821, in filter
    result = f.filter(record)
  File "/home/brendan/env/lib/python3.10/site-packages/sphinx/util/logging.py", line 433, in filter
    raise exc
sphinx.errors.SphinxWarning: /home/brendan/github/breathe/documentation/source/dot_graphs.rst:2:[Errno 2] No such file or directory: 'dotfile.dot'
```
</details>

I'm still trying figure out what it all means

---
FYI, the following info was copied from `sphinx-build -h`
```
-v                increase verbosity (can be repeated)
-W                turn warnings into errors
-E                don't use a saved environment, always read all files
```
The `-v` is a remnant from the previous usage (seen on master branch) `-vn`. However, `-n` will show warnings that we can't avoid.

I guess we can forego using `-E` (its a habit for me), but the `-W` was the whole point of this PR. This is why I want to get the builds working for you (it is your lib after all).

As stated in the PR description, using `-n -W` will cause the build to fail entirely. I tried fixing those "nit-picky" warnings, but there's ~100 or so, and I was more concerned with user experience (not dev completeness).

Reply:
@michaeljones You seem to have stumbled across a problem using dotfiles with doxygen v1.9..3. I re-ran the workflow (locally) with doxygen v1.9.2, and it completed with no errors. After check the change log of doxygen, this seems to be an intended change [stemming from this commit](https://github.com/doxygen/doxygen/commit/21323cd16d56124744f6b37283991af466dafa22). So far, setting `DOT_CLEANUP` doesn't seem to help with v1.9.3.

If it turns out that using a dotfile is going to be problematic, we can roll that back and prefer users use the `\dot` command instead (which doesn't suffer from path problems).

Reply:
@michaeljones The dotfile test should work now with doxygen 1.9.3. See [this commit comment](https://github.com/michaeljones/breathe/pull/816/commits/476f79c4206705648e2bd30f2a8d5a36a1f3e411) about the underlying problem.

Reply:
This was resolved by merging #821